pandas/core/generic.py

```diff
@@ -3915,6 +3915,9 @@ def drop(
 
         for axis, labels in axes.items():
             if labels is not None:
+                # Check for empty Index, GH 27994
+                if is_list_like(labels) and not len(labels):
```

Comment:
rather than do this, you can just handle the empty case in ``_drop_axis``

Reply:
Are you sure @jreback? I tried a little, but I couldn't get it work while making sure to raise `KeyError` when passing `str`/`List[str]` to `DataFrame.drop()` that has `DatetimeIndex`.

Reply:
I think the suggestion is to include the `not len(labels)` check in `NDFrame._drop_axis`. And when `labels` is empty we just return `self` from `_drop_axis`.

Reply:
can you try this